### DISCUSSION TITLE
PersianDateTimePicker

### DISCUSSION THREAD

**Discussion by md131376st** [HandyOrg/HandyControl#823] [Q&A] [answered]
Hi 
I want to use PersianDateTimePicker in my XAML file but the confirm button in the PersianDateTimePicker is not in Persian can you please help fix this issue.
what it looks like
![what it looks like ](https://user-images.githubusercontent.com/20892188/117890682-e4ff3300-b2ca-11eb-8e08-adfe2d31d9fd.PNG)
what I need it to look  like
![what I need it to look  like](https://user-images.githubusercontent.com/20892188/117890699-ea5c7d80-b2ca-11eb-8bf9-febbe19f89fb.PNG)

 ` <hc:PersianDateTimePicker
                    
                    Grid.Row="1"
                    Grid.Column="7"
                    Width="244"
                    Height="30"
                    FontFamily="/SnappFood;component/Resources/Fonts/#IRANSans Light"
                    Name="EndDatePicker" TextType="Persian" />`
I tried adding 
`HandyControl.Tools.ConfigHelper.Instance.SetLang("fa");` 
to my OnStartup function in App.xmal.cs file but it keeps getting System.Windows.Markup.XamlParseException.

**Comment by ghost1372** [comment, marked as answer]
You can use 
`ConfirmButtonText`
Property to change confirm button text

**Reply by md131376st** [reply]
can you please help with the correct xmal code?
I tried this but  get the property does not exist in xmal error.
` xmlns:hc="https://handyorg.github.io/handycontrol"`
`hc:PersianDateTimePicker
                    ConfirmButtonText ="تایید"
                    Grid.Row="1"
                    Grid.Column="7"
                    Width="244"
                    Height="30"
                    FontFamily="/SnappFood;component/Resources/Fonts/#IRANSans Light"
                    Name="EndDatePicker" TextType="Persian" />`

**Reply by ghost1372** [reply]
you should remove **TextType** property

**Reply by md131376st** [reply]
ConfirmButtonText this property is not in the PersianDateTimePicker

**Reply by ghost1372** [reply]
بله ببخشید حق با شماست این پراپرتی برای کنترل 
PersianCalendarWithClock
تعریف شده و داخل کنترل 
PersianDateTimePicker
من از 
PersianCalendarWithClock
استفاده کردم برای همین فراموش کردم دسترسی به این پراپرتی رو تعریف کنم امشب این مشکل رو حل میکنم و از طریق بیلد شبانه میتونید بروزرسانی رو دریافت کنید
اما اگر نمیخواید منتظر بمونید میتونید از روش زیر استفاده کنید
```
<StackPanel>
        <StackPanel.Resources>
            <Style TargetType="hc:PersianCalendarWithClock" BasedOn="{StaticResource {x:Type hc:PersianCalendarWithClock}}">
                <Setter Property="ConfirmButtonText" Value="تایید"/>
            </Style>
        </StackPanel.Resources>
        <hc:PersianDateTimePicker/>
    </StackPanel>
```

**Reply by md131376st** [reply]
ممنون

**Comment by ghost1372** [comment]
> Hi 
> I want to use PersianDateTimePicker in my XAML file but the confirm button in the PersianDateTimePicker is not in Persian can you please help fix this issue.
> what it looks like
> ![what it looks like ](https://user-images.githubusercontent.com/20892188/117890682-e4ff3300-b2ca-11eb-8e08-adfe2d31d9fd.PNG)
> what I need it to look  like
> ![what I need it to look  like](https://user-images.githubusercontent.com/20892188/117890699-ea5c7d80-b2ca-11eb-8bf9-febbe19f89fb.PNG)
> 
> 
>  ` <hc:PersianDateTimePicker
>                     
>                     Grid.Row="1"
>                     Grid.Column="7"
>                     Width="244"
>                     Height="30"
>                     FontFamily="/SnappFood;component/Resources/Fonts/#IRANSans Light"
>                     Name="EndDatePicker" TextType="Persian" />`
> I tried adding 
> `HandyControl.Tools.ConfigHelper.Instance.SetLang("fa");` 
> to my OnStartup function in App.xmal.cs file but it keeps getting System.Windows.Markup.XamlParseException.
> 
> 

Also please upload a sample project, it should not give you xamlparseexception error

**Reply by md131376st** [reply]
I changed the placement to after
` base.OnStartup(e);`
and it worked thank you for your help
